lib/versions.nix

```diff
@@ -61,4 +61,17 @@ rec {
     versionSuffix = lib.removePrefix numericVersion version;
   in lib.concatStringsSep "." (lib.take n (lib.splitVersion numericVersion ++ lib.genList (_: "0") n)) + versionSuffix;
 
+  /* Format date from YYYYMMDD to YYYY-MM-DD. This is usually used with a flake's input lastModifiedDate value.
+
+     Type: dateFormat :: String -> String
+
+     Example:
+       dateFormat "19700101"
+       => "1970-01-01"
+       dateFormat self.lastModifiedDate # (20230426163528)
+       => "2023-04-26"
+  */
+  dateFormat = d:
```

Comment:
This name is way too generic for such specific functionality. I'd call it something like `fromFlakesLastModifiedToISO8601`, or even better, change Flakes so it exposes this directly instead of having to parse a string

Reply:
Your suggested name is unacceptable in my opinion. Add this functionality to fetchTree/Flakes definitely seems better but as you can see there has been an issue for in Nix repo and since it wasn't picked up I decided to implement it as a library in nixpkgs instead.

Reply:
sourceInfo has `lastModified`, `lastModifiedDate`, `rev` and `shortRev`. What do you think it should be called?

Reply:
https://github.com/NixOS/nix/pull/8277